cmd/datablue/handlers.go

```diff
@@ -451,6 +453,72 @@ func varsHandler(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, resp)
 }
 
+// apiHandler handles API requests which take the form:
+//
+//	/api/operation/property/value
+func apiHandler(w http.ResponseWriter, r *http.Request) {
+	logRequest(r)
+
+	req := strings.Split(r.URL.Path, "/")
+	if len(req) < 5 {
+		writeError(w, errInvalidAPI)
+		return
+	}
+
+	var (
+		op   = req[2]
+		prop = req[3]
+		val  = req[4]
+	)
+	switch op {
+	case "test":
+		n, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			writeError(w, fmt.Errorf("could not parse size: %w", err))
+			return
+		}
+		body := make([]byte, n)
+
+		// Chunk size is optional.
+		chunk := n
+		if len(req) == 6 {
+			chunk, err = strconv.ParseInt(req[5], 10, 64)
+			if err != nil {
+				writeError(w, fmt.Errorf("could not parse chunk size: %w", err))
+				return
+			}
+		}
+
+		switch prop {
+		case "upload":
+			// Receive n bytes from the client.
+			_, err = io.ReadFull(r.Body, body)
+			if err != nil {
+				writeError(w, fmt.Errorf("could not read body: %w", err))
+				return
+			}
+			fmt.Fprint(w, "OK")
+			return
+
+		case "download":
+			// Send n bytes to the client.
+			h := w.Header()
+			h.Add("Content-Type", "application/octet-stream")
+			h.Add("Content-Disposition", "attachment; filename=\""+val+"\"")
+			rand.Read(body)
+			var i int64
+			for i = 0; i < n; i += chunk {
+				w.Write(body[i : i+chunk])
+			}
+			return
+		}
+
```

Comment:
Do we know what the future api routes might look like? Is it worth adding in extra cases which return "not yet implemented"